From 2cce74e5174caa52bd5bef1552000081a246de43 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Wed, 17 Apr 2024 15:19:54 +0200
Subject: [PATCH 589/954] feat(st): remove unused IWDG flags

Remove flags used for configuring watchdog during low power phases.
The configuration to stop watchdog during standby and s2idle is done in
OP-TEE.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 82a242f1c5d8c46078a9e0b4035f67a0867e9f47 feat(st): remove unused IWDG flags]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: Ib68d5bf31d7f97791af6b25327b374771b75fd4e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/374301
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/st/iwdg/stm32_iwdg.c    | 25 +------------------------
 include/drivers/st/stm32_iwdg.h |  2 --
 2 files changed, 1 insertion(+), 26 deletions(-)

diff --git a/drivers/st/iwdg/stm32_iwdg.c b/drivers/st/iwdg/stm32_iwdg.c
index 9fd09504f3..8ec88beea3 100644
--- a/drivers/st/iwdg/stm32_iwdg.c
+++ b/drivers/st/iwdg/stm32_iwdg.c
@@ -33,7 +33,6 @@
 struct stm32_iwdg_instance {
 	uintptr_t base;
 	unsigned long clock;
-	uint8_t flags;
 	int num_irq;
 };
 
@@ -112,18 +111,7 @@ int stm32_iwdg_init(void)
 		iwdg->base = dt_info.base;
 		iwdg->clock = (unsigned long)dt_info.clock;
 
-		/* DT can specify low power cases */
-		if (fdt_getprop(fdt, node, "stm32,enable-on-stop", NULL) ==
-		    NULL) {
-			iwdg->flags |= IWDG_DISABLE_ON_STOP;
-		}
-
-		if (fdt_getprop(fdt, node, "stm32,enable-on-standby", NULL) ==
-		    NULL) {
-			iwdg->flags |= IWDG_DISABLE_ON_STANDBY;
-		}
-
-		/* Explicit list of supported bit flags */
+		/* Read OTP to know if IWDG is started by hardware or not */
 		hw_init = stm32_iwdg_get_otp_config(idx);
 
 		if ((hw_init & IWDG_HW_ENABLED) != 0) {
@@ -132,7 +120,6 @@ int stm32_iwdg_init(void)
 				      idx + 1U);
 				panic();
 			}
-			iwdg->flags |= IWDG_HW_ENABLED;
 		}
 
 		if (dt_info.status == DT_DISABLED) {
@@ -141,14 +128,6 @@ int stm32_iwdg_init(void)
 			continue;
 		}
 
-		if ((hw_init & IWDG_DISABLE_ON_STOP) != 0) {
-			iwdg->flags |= IWDG_DISABLE_ON_STOP;
-		}
-
-		if ((hw_init & IWDG_DISABLE_ON_STANDBY) != 0) {
-			iwdg->flags |= IWDG_DISABLE_ON_STANDBY;
-		}
-
 		VERBOSE("IWDG%u found, %ssecure\n", idx + 1U,
 			((dt_info.status & DT_NON_SECURE) != 0) ?
 			"non-" : "");
@@ -162,8 +141,6 @@ int stm32_iwdg_init(void)
 		if ((hw_init & IWDG_HW_ENABLED) == 0) {
 			/* Use default timeout, ignore DT's "timeout-sec" */
 			stm32_iwdg_start(iwdg);
-
-			iwdg->flags |= IWDG_HW_ENABLED;
 		}
 	}
 
diff --git a/include/drivers/st/stm32_iwdg.h b/include/drivers/st/stm32_iwdg.h
index bad25244a8..a69701a764 100644
--- a/include/drivers/st/stm32_iwdg.h
+++ b/include/drivers/st/stm32_iwdg.h
@@ -10,8 +10,6 @@
 #include <stdint.h>
 
 #define IWDG_HW_ENABLED			BIT(0)
-#define IWDG_DISABLE_ON_STOP		BIT(1)
-#define IWDG_DISABLE_ON_STANDBY		BIT(2)
 
 int stm32_iwdg_init(void);
 void stm32_iwdg_refresh(void);
-- 
2.53.0

